crates/walrus-service/src/node/storage.rs

```diff
@@ -668,7 +668,7 @@ impl Storage {
                 }
             };
 
-            if blob_info.is_registered(current_epoch) {
+            if !blob_info.can_data_be_deleted(current_epoch) {
                 tracing::trace!(
                     %blob_id,
                     "skipping blob that is still registered",
```

Comment:
nit: also update the doc to mention about `deletion`, and disconnect it wit `register`?